client/src/js/services/TransactionTypeStoreService.js

```diff
@@ -0,0 +1,43 @@
+angular.module('bhima.services')
+.service('TransactionTypeStoreService', TransactionTypeStoreService);
+
+// dependencies injection
+TransactionTypeStoreService.$inject = ['$q', 'TransactionTypeService', 'Store'];
+
+/**
+ * Transaction Type Store Controller
+ */
+function TransactionTypeStoreService($q, TransactionType, Store) {
+  'use strict';
+
+  var service = this;
+  var data = new Store();
+  var initialLoad = true;
+
+  var request = TransactionType.read()
+    .then(function (result) {
+      data.setData(result);
+      initialLoad = false;
+      return data;
+    });
+
+  service.data = transactionTypeStore;
+  service.refresh = refresh;
+
+  function refresh() {
+    TransactionType.read()
+    .then(function (result) {
+      data.setData(result);
+    });
+  }
+
+  function transactionTypeStore() {
+    if (initialLoad) {
+      return request.then(function (result) {
```

Comment:
This code is redundant.
```js
return request.then(function (result) { return result; });
// is the same as
return request;
```
There is no need to have the `.then()` statement.

This function could be written as:
```js
function transactionTypeStore() {
  return initialLoad ? request : $q.resolve(data);
}
```
(but you would need comments describing what it is doing)